docs/DeveloperGuide.md

```diff
@@ -6,32 +6,215 @@
 
 ## Design & implementation
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### Implemented Features
 
+Following a restucture to the project to follow a more Object Oriented structure, as well as to allow for conformity and easy editing of code, many of our commands will follow a similar structure of the following:
+
+1. User enters a command, which is given to a Command Handler
+2. The handler passes this to a corresponding Parser object for that command, which will split it into its necessary parts
+3. The Parser will create a corresponding Command object for that command, which will perform its necessary actions
+4. The Command object will return successfully to the Command Handler, and the output will be given to the user (output for the user is stored in a `Response` class)
+
+The command to add a project is one example of this structure:
+
+#### Add a Project
+![image info](./UmlDiagrams/addProjectNew.png)
```

Comment:
1. Possible to remove the arrow that has the name "return" on it
![image](https://user-images.githubusercontent.com/69310031/160963240-0064cdaf-92e9-4c96-8ded-07bfec33f13b.png)

2. The arrow pointing from `Command Handler` self-invocation activation bar should touch the activation bar of `DeleteProjectCommand`.
![image](https://user-images.githubusercontent.com/69310031/160963261-39bf6aac-6a29-4f26-b3ab-9dd91d5de062.png)

Reply:
I second this change. The return arrow should be removed. Furthermore, I think some diagramming tool can be used to make the lines straighter.